autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -44,42 +62,86 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
+        if (DELETE_COMPLETED.equals(stackResponse.getStatus())) {
             beforeDeleteCleanup(cluster);
             clusterService.removeById(cluster.getId());
-            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), statusResponse.getStatus());
+            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), stackResponse.getStatus());
         } else if (clusterAvailable && !RUNNING.equals(cluster.getState())) {
             clusterService.setState(cluster.getId(), ClusterState.RUNNING);
             LOGGER.info("Updated cluster '{}' to Running, CB Stack Status '{}', CB Cluster Status '{}'.",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+                    cluster.getStackCrn(), stackResponse.getStatus(), stackResponse.getCluster().getStatus());
         } else if (!clusterAvailable && RUNNING.equals(cluster.getState())) {
-            clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
-            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+            clusterService.setState(cluster.getId(), SUSPENDED);
+            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'", cluster.getStackCrn(), stackResponse.getStatus(),
+                    stackResponse.getCluster().getStatus());
         } else if (RUNNING.equals(cluster.getState()) && (cluster.getMachineUserCrn() == null || cluster.getEnvironmentCrn() == null)) {
             populateEnvironmentAndMachineUserIfNotPresent(cluster);
         }
     }
 
+    private boolean determineClusterAvailability(StackV4Response stackResponse, DependentHostGroupsV4Response dependentHostGroupsResponse,
+            Set<String> policyHostGroups) {
+        if (undefinedDependencyPresent(dependentHostGroupsResponse, policyHostGroups)) {
+            return stackResponse.getStatus().isAvailable();
+        }
+
+        boolean dependentHostsUnhealthy = policyHostGroups.stream().anyMatch(hg -> dependentHostsUnHealthy(dependentHostGroupsResponse, stackResponse, hg));
+
+        return !(stackDeletionInProgress(stackResponse) || stackModificationInProgress(stackResponse)
+                || dependentHostsUnhealthy);
+    }
+
+    private boolean dependentHostsUnHealthy(DependentHostGroupsV4Response dependentHostGroupsResponse, StackV4Response stackResponse, String policyHostGroup) {
+        return stackResponse.getInstanceGroups().stream()
+                .flatMap(ig -> ig.getMetadata().stream())
+                .filter(im -> dependentHostGroupsResponse.getDependentHostGroups().getOrDefault(policyHostGroup, emptySet())
+                        .contains(im.getInstanceGroup()))
+                .anyMatch(im -> !InstanceStatus.SERVICES_HEALTHY.equals(im.getInstanceStatus()));
```

Comment:
Can we capture all, and generate a log message when there are unhealthy hosts. Good to capture this in the logs.

Next question: How do we surface this to the customer. Maybe an event on the Activity Console indicating that AutoScale disabled because X nodes are unhealthy, etc (ONLY for the case where it is disabled for unhealthy nodes).
The ActivityConsole could be a follow up jira - but would be an important one.

Reply:
Addressed this. Collecting unhealthy hostNames and logging them if present. 
Also, filed https://jira.cloudera.com/browse/CB-19387 to track their propagation to the activity console